protocol/listgroups/listgroups.go

```diff
@@ -57,8 +59,15 @@ func (r *Response) Merge(requests []protocol.Message, results []interface{}) (
 	response := &Response{}
 
 	for r, result := range results {
-		brokerResp := result.(*Response)
-		respGroups := []ResponseGroup{}
+		brokerResp, ok := result.(*Response)
```

Comment:
@kscooo good catch! What do you think about modeling your bug fix after this one? https://github.com/segmentio/kafka-go/blob/da91759e946436612f6e31af58971162bf1d058d/protocol/listoffsets/listoffsets.go#L151

Reply:
forgive me for not knowing the rest of the source code for this project, it looks like this fix is much better

Reply:
No problem! The `protocol` package (`"github.com/segmentio/kafka-go/protocol"`) already has an implementation of the `Result` function though, do you think you could use that instead of redefining it in this file?

Reply:
Removed, sorry for not noticing that the `*Response` implements `Message` interface